external-crates/move/crates/move-regex-borrow-graph/src/regex.rs

```diff
@@ -0,0 +1,343 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::fmt::{Debug, Display};
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) struct Regex<Lbl> {
+    labels: Vec<Lbl>,
+    ends_in_dot_star: bool,
+}
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) enum Extension<Lbl> {
+    Epsilon,
+    Label(Lbl),
+    DotStar,
+}
+
+//**************************************************************************************************
+// impls
+//**************************************************************************************************
```

Comment:
I would love a small semantic description of regex operations as a comment at the top here -- just for sanity later